File: wraplayout/src/main/java/autoexclue/Segment.java
package autoexclue;

import java.util.ArrayList;

public class Segment {
    int layoutRows;
    ArrayList<Integer> childMaxWidth;
    int measureRows;
    boolean haveMeasured;

    Segment(int start,
            int end,
            int size) {
        this.start = start;
        this.end = end;
        this.size = size;
    }

    int start;
    int end;
    int size;
    int height;
    int width;

    public void reset() {
        start = 0;
        end = 0;
        size = 0;
        height = 0;
        width = 0;
    }
}

File: wraplayout/src/main/java/autoexclue/Segment.kt
package autoexclue

data class Segment constructor(
        var start: Int = 0,
        var end: Int = 0,
        var size: Int = Math.max(0, end - start),
        var layoutRows: Int = 0,
        var height: Int = 0,
        var width: Int = 0,
) {
    override fun toString(): String {
        return "[$start~$end=$size]"
    }
}

fun Segment?.rangeEquals(other: Segment?): Boolean {
    if (this == other) return true
    if (other == null || this == null) return false
    return this.start == other.start && this.end == other.end && this.size == other.size
}

fun Segment.reset() {
    this.start = 0
    this.end = 0
    this.size = 0
    this.height = 0
    this.width = 0
}